compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -961,13 +961,20 @@ fn check_param_wf(tcx: TyCtxt<'_>, param: &hir::GenericParam<'_>) -> Result<(),
                         hir_ty.span,
                         "using raw pointers as const generic parameters is forbidden",
                     ),
-                    _ => tcx.dcx().struct_span_err(
-                        hir_ty.span,
-                        format!("`{}` is forbidden as the type of a const generic parameter", ty),
-                    ),
+                    _ => {
+                        // Avoid showing "{type error}" to users. See #118179.
+                        ty.error_reported()?;
+
+                        tcx.dcx().struct_span_err(
+                            hir_ty.span,
+                            format!(
+                                "`{ty}` is forbidden as the type of a const generic parameter",
+                            ),
+                        )
+                    }
                 };
 
-                diag.note("the only supported types are integers, `bool` and `char`");
+                diag.note("the only supported types are integers, `bool`, and `char`");
```

Comment:
comma before `and` looks wrong to me.

Reply:
It's called the oxford comma.

Reply:
Hmm, okay, looking at *.stderr there used both cases with and w\o comma with enumerations.

Reply:
Found only 2 mentions: https://github.com/rust-lang/rustc-dev-guide/pull/1087#discussion_r593821973, https://github.com/rust-lang/rustc-dev-guide/pull/1139#discussion_r650445567; should be added to the guide?